### PR TITLE
Add support for IPv6 addresses enclosed in square brackets in the destination parser.

### PR DESCRIPTION
**Pull request by SingularityCat**
This should (partially) address issue #418.

### CODE REVIEW

File: lib/carbon/util.py [hunk 1]
```diff
@@ -100,15 +100,23 @@ def parseDestinations(destination_strings):
...
 
   for dest_string in destination_strings:
-    parts = dest_string.strip().split(':')
-    if len(parts) == 2:
-      server, port = parts
-      instance = None
-    elif len(parts) == 3:
-      server, port, instance = parts
+    s = dest_string.strip()
+    bidx = s.rfind("]:")    # find closing braket and following colon.
```
Comment (obfuscurity) [inline lib/carbon/util.py]: Typo in comment, should be `bracket`.

**Comment by SingularityCat** [comment]
Like this? :smile_cat:

**Comment by obfuscurity** [comment]
Yes, thanks!

**Comment by obfuscurity** [comment]
@iain-buclaw-sociomantic @iksaif Any objections?

**Comment by iain-buclaw-sociomantic** [comment]
Could a test be added?  Not really a blocker as I'm sure that the carbon tests could do with some TLC.

**Comment by SingularityCat** [comment]
I did look to see if there were tests for utils.py, but I couldn't find any - it seems to be incompletely tested by virtue of being a dependency. I'd happily write a few unit tests for parseDestinations though. Would a new 'test_util.py' be fine?

**Comment by SingularityCat** [comment]
Got a bit sidetracked... Anyway, hope these cover enough.

File: lib/carbon/util.py [hunk 2]
```diff
@@ -100,15 +100,23 @@ def parseDestinations(destination_strings):
   destinations = []
 
   for dest_string in destination_strings:
-    parts = dest_string.strip().split(':')
-    if len(parts) == 2:
-      server, port = parts
-      instance = None
-    elif len(parts) == 3:
-      server, port, instance = parts
+    s = dest_string.strip()
```
Comment (iksaif) [inline lib/carbon/util.py]: Could this be moved into a specific function ? (to make it easier to use ipaddress module later)

Comment (iksaif) [inline lib/carbon/util.py]: anything would be fine :)

**Comment by iksaif** [comment]
LGTM

**Pull request merged**